src/Autocomplete/AutocompleteOverlay.tsx

```diff
@@ -49,6 +49,8 @@ function AutocompleteOverlay({
     return null
   }
 
+  if (!showMenu) return null
```

Comment:
Hi! 

Given that Overlay is part of the [public API](https://primer.style/react/Overlay/) and it accepts a `visibility` prop, should the logic of attaching listeners based on visibility be inside Overlay instead?

Reply:
Hey @siddharthkp, good question! I had originally tried to fix this bug through the `Overlay`, but essentially, `Overlay` doesn't always have a dropdown menu, but `Autocomplete` does. To implement this fix for the case where an  `Autocomplete` is contained in an `Overlay` (where the dropdown menu needs to behave as it already does) we would need to pass along a ref up to the parent, or create a context to do this, amongst other things. After spending some time going down that rabbit hole, your single line fix seemed to make a lot more sense! It does make sense to use the `showMenu` variable that already exists instead of trying to recreate it elsewhere.

Also, `visibility` in `Overlay` still needs to as it already does. We're using that every time in a `useLayoutEffect` hook, which is to make the animation work properly. The `AutoComplete` dropdown menu shouldn't effect that, it just needs to be the focus of the key event when it's being interacted with.